Fixity values from Arkivum

Hwesta opened the pull request:
- Add Space-specific fixity hook
- Add uncompressed AIP support to Arkivum with new `api/3/ingest-manifest/` endpoints
- Add checking if a file is local for uncompressed bags in Arkivum

Based on top of #88

Hwesta committed:
Add structure for Space-specific check_fixity

Hwesta committed:
API: Download package always gives mimetype & attachment

Removed different behaviour if package is a 7z or zip. All downloads guess
mimetype, and have a suggested filename in content-disposition.

Hwesta committed:
Fixtures: Clean up to work better with other usecases

Hwesta committed:
API: Fully qualify TastyPie exception being raised

Hwesta committed:
Arkivum: Add function to check if a package is locally available

Arkivum files are NFS-mounted and have local paths but may not be
immediately available locally. Add a function to check and return the
status of a package.

Hwesta committed:
API: Return Arkivum-specific status codes

Before fetching an Arkivum file (including checking if it's compressed)
check if the file is locally available and return an error if it isn't.

Hwesta committed:
Arkivum: Add uncompressed AIP support

Update post_move_from_storage_service and update_package_status to call the
new bag-only endpoint for an uncompressed package.

Hwesta committed:
Arkivum: add fixity support

For uncompressed bags, return fixity info from package status.

mistydemeo commented on storage_service/locations/models/arkivum.py:
```diff
@@ -123,42 +131,145 @@ def post_move_from_storage_service(self, staging_path, destination_path, package
+        :return: Tuple of (success, [errors], message)
+        """
+        if package.is_compressed:
+            return (None, [], 'Package is not a bag in Arkivum.')
```
Is this message accurate?

Hwesta committed:
Arkivum: Check if a file in uncompressed packages is local

Hwesta committed:
HACK remove this - disable SSL checking

mistydemeo commented:
:+1:

Hwesta committed:
Arkivum: Add log for is_local error

Check for local status is bagit call to Arkivum. Log error if missing.

Hwesta committed:
Arkivum: Check single file local for package

Hwesta committed:
WIP allow HEAD to download file/package

Hwesta committed:
Arkivum: Send email if file is not local

Hwesta committed:
Arkivum: Move non-local email to Arkivum

Move sending an email to superusers if the file is not cached in Arkivum to
the is_local function. Also include the ID from the Arkivum response.

Hwesta committed:
Arkivum: Only email if not cached and GET request

Hwesta committed:
Arkivum: fixity scheduled is not success

Hwesta committed:
Fixity: return timestamp of last run

Hwesta committed:
Arkivum: Fixity only passes if replication is green

Update fixity to only return success if the fixity passed and the
replication status is green. If the fixity completed but the status is
amber, or the fixity is still scheduled, then the status of the files
can't be confirmed.

Hwesta committed:
Fixity: Add force_local to endpoint

Rename ignore_space to force_local, and add as an option to the API endpoint.

Hwesta committed:
Fixity: Only return space timestamp

Only return the space's timestamp, don't generate a local timestamp. An
external API can generate a timestamp or duration based on when the call
happened.

Timestamps are always returned in ISO format.

Hwesta committed:
fixup! Fixity: return timestamp of last run

sevein committed:
utils: serve files with http.FileResponse

FileResponse is a subclass of StreamingHttpResponse optimized for binary files.
It uses wsgi.file_wrapper if provided by the wsgi server, otherwise it streams
the file out in small chunks.

Hwesta commented:
Replaced by #135 which also includes #88 and cleanups.
